### DISCUSSION TITLE
Crazyflie 2.1 only executes take-off once when using Python API scripts

### DISCUSSION THREAD

**Discussion by rens-van-vendeloo** [bitcraze/discussions#758] [Q&A]
I am currently exploring the capabilities of the CrazyFlie 2.1 in combination with the CrazyRadio. When I execute the example code given in Step 2 of [Step-By-Step: Motion Commander](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_motion_commander/#step-2-take-off-function) the crazyflie executes this perfectly. However if I try to run the script again, it only gives the logging output for the deck detection and it does not take off. This is fixed when I restart the drone.

Currently I cannot succesfully execute a script twice, without restarting the drone in between each execution. Any ideas how to fix this or what the issue is?

Thanks in advance!

**Comment by cyborgdennett** [comment]
I came across this aswell, please fix!

**Comment by knmcguire** [comment]
Hi! Then probably there is something going on with the parameter callback. What kind of print out do you get? Does it say that it doesn't have the flowdeck attached?

**Reply by rens-van-vendeloo** [reply]
Hi when I run the python script, it says:

1
Deck is attached!

I hope this helps, if you need more information, let me know.

**Comment by kulasdrew** [comment]
We are having the same issue.  We found that when we run the example script once, even though it connects it does not properly execute the code.  It will run through all the callbacks and output what we expect at each step but we get zero response from the motors.  Also, every time we call it the connection light turns on and turns off after outputting everything we expect.  We have tried it with both the CrazyRadio PA and 2.0 with identical behavior.  (When we un-plug the drone and plug it back in we can run the script once and the motors will spin but it only runs once) We have no decks attached.  

Here is the code we are running:

```python
import cflib
import logging 
from cflib.crazyflie.syncCrazyflie import SyncCrazyflie
from cflib.positioning.motion_commander import MotionCommander 

from cflib.utils import uri_helper

uri = uri_helper.uri_from_env(default='radio://0/80/2M/E7E7E7E7E7')

logging.basicConfig(level=logging.ERROR)

if __name__ == '__main__':
    cflib.crtp.init_drivers()
    with SyncCrazyflie(uri) as scf:
        with MotionCommander(scf) as mc:
            mc.forward(1)
```

**Reply by kulasdrew** [reply]
Our solution was to downgrade the firmware from 2023.06 to 2023.02 on the Crazyflie and we are able to run programs multiple times uninterrupted.

**Comment by krichardsson** [comment]
[We did an update of the tumble detection](https://github.com/bitcraze/crazyflie-firmware/pull/1286) that we have seen is perhaps a bit too sensitive. The tumble detection tries to detect if the Crazyflie is upside down, stuck in some obstacle or similar, but it seems as it is also triggered sometimes when the Crazyflie makes a hard landing (it is only  using the accelerometers for detection). When a tumble is detected in flight, the system will turn off the motors and lock and requires a reboot to fly again. I think this might be what you see.

The problem is documented in https://github.com/bitcraze/crazyflie-firmware/issues/1294

A workaround (requires rebuild of the firmware) is to set the values as indicated in the issue. If anyone tries this, please let us know if it solved the problem. Tweaking parameters like this is hard to test for all situations and environments!
